futures-util/src/stream/forward.rs

```diff
@@ -90,8 +84,9 @@ where
                 Poll::Ready(Some(Err(e))) => return Poll::Ready(Err(e)),
                 Poll::Ready(None) => {
                     try_ready!(self.as_mut().sink().as_pin_mut().expect(INVALID_POLL)
-                                   .poll_close(lw));
-                    return Poll::Ready(Ok(self.as_mut().sink().take().unwrap()))
+                                   .poll_flush(lw));
```

Comment:
This should be back to `poll_close`, right?

Reply:
If I switch this back to `poll_close()`, the example code in the repo on #1445 doesn't work as expected.

Reply:
When I switch this back to `poll_close()`, then the use of `forward()` in the example code shown in #1440 no longer works as expected.

Reply:
Thanks for merging #1443, @cramertj! I've switched back to `poll_close()`, as I can confirm on my machine that `StreamExt::forward()` now behaves as expected, identically to 0.1.25.